scripts/base/frameworks/analyzer/dpd.zeek

```diff
@@ -120,7 +120,16 @@ event analyzer_violation_info(atype: AllAnalyzers::Tag, info: AnalyzerViolationI
 	if ( ignore_violations_after > 0 && size > ignore_violations_after )
 		return;
 
-	disable_analyzer(c$id, aid, F);
+	local disabled = disable_analyzer(c$id, aid, F);
+
+	# add "-service" to the list of services on removal due to violation, if analyzer was confirmed before
+	if ( track_removed_services_in_connection && disabled && Analyzer::name(atype) in c$service )
+		{
+		local rname = fmt("-%s", Analyzer::name(atype));
```

Comment:
`cat("-", name(atype))` or event `"-" + ...` is likely faster than the format string machinery if you wanted to replace that.